src/Voron/Platform/Win32/WindowsMemoryMapPager.cs

```diff
@@ -66,6 +61,8 @@ public struct SplitValue
             FileName = file;
             _logger = LoggingSource.Instance.GetLogger<StorageEnvironment>($"Pager-{file}");
 
+            _canPrefetchAhead = !fileAttributes.HasFlag(Win32NativeFileAttributes.Temporary);
```

Comment:
I was a bit confused myself for a while. So let me try to explain. It is not that we cannot prefetch on temporary files, we cannot prefetchAhead which is an entirely different thing; the standard usage prefetch works on temporary files too. 

The underlying problem is that we don't have a unified view of the 'page translation' at all moments. 

So whenever we have a page number it may happen that:
- It is in the proper location on the data file (we can prefetch ahead those)
- It has been modified and written to the journal. 
- It is in the scratchpad (that is hot anyway)

Prefetching ahead assumes that the page number corresponds directly on a memory position, and that is why you prefetch, therefore no translation has to be done. If we would like to prefetch ahead, we need to know exactly where the pages are stored (which probably is not contiguously anyways). To make it work we must do the page translation but that information is not available at the time we want to do the prefetch. 

So, in essence, to prefetch ahead from a Journal is not such a good idea in general because it assumes that the data layout is already optimized. So instead of trying to page translate every entry in the tree, we just let the dynamic prefetching to take care of it and ignore any prefetch ahead request happening to end into a temporary map file.

Reply:
Moreover, the usage pattern of the prefetch ahead is especially useful when we are just starting. Which also has the side effect to disabling itself by design when the database has moved into a high frequency write mode.